src/main/webapp/app/shared/exercise-filter/exercise-filter-modal.component.ts

```diff
@@ -0,0 +1,194 @@
+import { Component, EventEmitter, OnInit, Output, ViewChild } from '@angular/core';
+import { NgbActiveModal, NgbTypeahead } from '@ng-bootstrap/ng-bootstrap';
+import { faFilter } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+import { FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { SidebarCardElement, SidebarData } from 'app/types/sidebar';
+import { Observable, OperatorFunction, Subject, merge } from 'rxjs';
+import { debounceTime, distinctUntilChanged, filter, map } from 'rxjs/operators';
+import { CustomExerciseCategoryBadgeComponent } from 'app/shared/exercise-categories/custom-exercise-category-badge.component';
+import { RangeSliderComponent } from 'app/shared/range-slider/range-slider.component';
+import {
+    DifficultyFilterOption,
+    ExerciseCategoryFilterOption,
+    ExerciseFilterOptions,
+    ExerciseFilterResults,
+    ExerciseTypeFilterOption,
+    FilterOption,
+    RangeFilter,
+} from 'app/types/exercise-filter';
+import { satisfiesFilters } from 'app/shared/exercise-filter/exercise-filter-modal.helper';
+import { DifficultyLevel, ExerciseType } from 'app/entities/exercise.model';
+import { ExerciseCategory } from 'app/entities/exercise-category.model';
+
+export type FilterDetails = {
+    searchedTypes: ExerciseType[] | undefined;
+    selectedCategories: ExerciseCategory[];
+    searchedDifficulties: DifficultyLevel[] | undefined;
+    isScoreFilterApplied: boolean;
+    isPointsFilterApplied: boolean;
+    achievedScore: RangeFilter | undefined;
+    achievablePoints: RangeFilter | undefined;
+};
```

Comment:
Can we not define undefined variables with `?` like this:
```suggestion
export type FilterDetails = {
    searchedTypes?: ExerciseType[];
    selectedCategories: ExerciseCategory[];
    searchedDifficulties?: DifficultyLevel[];
    isScoreFilterApplied: boolean;
    isPointsFilterApplied: boolean;
    achievedScore?: RangeFilter;
    achievablePoints?: RangeFilter;
};
```